### PR TITLE
HIVE-27876 Incorrect query results on tables with ClusterBy & SortBy

### PR DESCRIPTION
**Pull request by ramesh0201**
### What changes were proposed in this pull request?
Disable `hive.map.groupby.sorted` optimization.

### Why are the changes needed?
The optimization leads to data correctness issues. See jira https://issues.apache.org/jira/browse/HIVE-27876 for details

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/GroupByOptimizer.java [hunk 1]
```diff
@@ -28,19 +28,11 @@
 import java.util.Set;
 import java.util.Stack;
 
+import org.apache.hadoop.hive.ql.exec.*;
```
Comment (kasakrisz) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/GroupByOptimizer.java:31]: AFAIK Hive code style doesn't allow wild card imports.
Here is some discussion why https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

Please revert this change.

**Review by kasakrisz — CHANGES_REQUESTED** [review]

**Comment by kasakrisz** [comment]
@ramesh0201 
There are a few tests for `GroupByOptimizer` named groupby_sort_*.q
Most of them has a step 
```
-- perform an insert to make sure there are 2 files
```
https://github.com/apache/hive/blob/7638cb1a3b07713cc490aa2909a37037f89e08b4/ql/src/test/queries/clientpositive/groupby_sort_3.q#L12-L13

However when I run these locally and monitored the file system I experienced that only one file remained after executing this step so there might be hidden data correctness issues in those cases too.

**Comment by aturoczy** [comment]
I don't get it. So if the CLUSTER BY and SORT BY are in the same command the result is incorrect? In which scenario it needed for anybody in the same statement?

**Comment by aturoczy** [comment]
I don't think this patch is correct. Please revisit this implementation.

**Review by aturoczy — CHANGES_REQUESTED** [review]
-1

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/GroupByOptimizer.java (hunk 1 above)
Comment (ramesh0201) [reply ql/src/java/org/apache/hadoop/hive/ql/optimizer/GroupByOptimizer.java:31]: My bad, I missed this.

**Comment by ramesh0201** [comment]
Thanks for the review @kasakrisz and @aturoczy. Let me rework on this.

**Review by kasakrisz — APPROVED** [review]

**Comment by kasakrisz** [comment]
Pending tests
